service/storage_service.cc

```diff
@@ -3900,9 +3900,9 @@ future<> storage_service::on_change(inet_address endpoint, application_state sta
             slogger.debug("Ignoring state change for dead or unknown endpoint: {}", endpoint);
             co_return;
         }
+        slogger.debug("endpoint={} on_change:     updating system.peers table", endpoint);
+        co_await do_update_system_peers_table(endpoint, state, value);
```

Comment:
Yeah, I need to expand on the motivation.
Since @asias insists on not initializing the peer status (to "shutdown")
the peers that are down after full cluster restart never become normal token owners,
hence this change.

> system.peers is supposed to contain only normal nodes.

Is it?
I think it contains nodes while they are leaving too.

It's be easier if we kept the peer state there too
and manage the table in a distributed way over raft :)

> Aren't you turning bootstrapping nodes into normal nodes prematurely this way?

I need to swap this patch with the next one:
`storage_service: on_join: defer on_change for STATUS and other states`
so that the status will be handled last and saved only if the endpoint is a normal token owner is we do today.

I think it should ok to have endpoints in system.peers with no STATUS while they bootstrap, but I don't know if this is cleaned up if bootstrapping failed midway.
@asias, @tgrabiec what do you think?

Reply:
Turns out we can do without this patch.
I'll drop it in the next version.

Reply:
But you got a good point about adding the state prematurally.
It interferes with replace, as seen in the `test_replace_node_no_hibernate_state` dtest.

Reply:
I dropped this logic from the next version.
Instead, the endpoint state of the explicitly ignored nodes will be used in prepare_replacement_info to populate topology and token_metadata